Unbound: Added ability to disable automatically added host entries

sestary opened the pull request:
I needed to disable the host entries automatically enabled by pfSense since my zone is DNSSEC signed and this record would not be returned with DNSSEC, the clients that validated DNSSEC would not be able to resolve the DNS entry.

Found that there was another ask on the forum for this:

https://forum.pfsense.org/index.php?topic=110970.0

Since this is my first pull request let me know if it is not done correctly.

Thank you,

Robbert

sestary commented:
I created feature #7525 on Redmine to track this.

pfSense-git-updates commented:
Before this pull request can be accepted you must first sign a CLA as described at https://www.pfsense.org/about-pfsense/#cla. Please read for more details.

sestary commented:
Just signed the CLA.

Also was just taking a look at the functions that generate the system host entries and I am wondering if it is better to add an option to the general settings page, under the "System" options that disables only the local entries from being created if it is set.

`/* Concatenate local, dnsmasq/unbound and dhcpd/dhcpdv6 hosts entries */
function system_hosts_entries($dnscfg) {
	$local = system_hosts_local_entries();
`

This would be more flexible since right now the code in the pull request stops any manually added host entries from working and this would disable it for all the other software too. I was trying to keep it in unbound specifically but now that I am thinking about it could be better as a more general option.

jim-p commented on src/usr/local/www/services_unbound_advanced.php:
```diff
@@ -350,6 +360,14 @@
+	'Disable Auto-added Host Entries',
+	'Disable the automatically-added host entries',
+	$pconfig['disable_auto_added_host_entries']
+))->setHelp('By default, the primary IPv4 and IPv6 addresses of the pfsense machine are added as records for the pfSense system domain (System | General Setup | Domain). ' .
```
Change "of the pfsense machine" to "of this firewall"
Change "for the pfSense system domain" to "for the system domain of this firewall"
Remove " (System | General Setup | Domain)" or change to "(System > General Setup)"

sestary commented:
I updated the help entries and also changed the way the local host entries are disabled. This allows manually defined host overrides to continue to work and also allows this option to be easily added to the DNSMasq page if someone wants it. I did not add it to DNSMaq since there is no advanced page and I wasn't sure it should clog up the main page.

Thanks Robbert

jim-p commented on src/usr/local/www/services_unbound_advanced.php:
```diff
@@ -364,8 +364,7 @@
 	$pconfig['disable_auto_added_host_entries']
-))->setHelp('By default, the primary IPv4 and IPv6 addresses of the pfsense machine are added as records for the pfSense system domain (System | General Setup | Domain). ' .
-			'This disables the auto generation of these entries.');
+))->setHelp('By default, the primary IPv4 and IPv6 addresses of of this firewall are added as records for the system domain of this firewall as configured in %1$sSystem: General Setup%2$s. This disables the auto generation of these entries.', '<a href="system.php">', '</a>');
```
Duplicated word "addresses **of of** this firewall"

sestary replied:
Fixed. I could have sworn I looked at this line 10 times to ensure I had no other typos.
